File: Python/Program/ParkingLot.py
class ParkingLot:
    def __init__(self, kapasitas, jml_kendaraan):
        """
        Inisialisasi objek ParkingLot dengan kapasitas dan jumlah kendaraan.

        Parameters:
        kapasitas (int): Kapasitas gedung parkir.
        jml_kendaraan (int): Jumlah kendaraan yang saat ini terparkir.
        """
        self.__kapasitas = kapasitas
        self.__jml_kendaraan = jml_kendaraan

    def add_vehicle(self, vehicle):
        """
        Menambahkan kendaraan ke dalam gedung parkir.

        Parameters:
        vehicle: Objek kendaraan yang akan ditambahkan.
        """
        self.__jml_kendaraan += 1

    def display_info(self):
        """
        Menampilkan informasi tentang gedung parkir, termasuk kapasitas dan jumlah kendaraan.
        """
        print(f"Kapasitas Gedung Parkir: {self.__kapasitas}")
        print(f"Jumlah Kendaraan: {self.__jml_kendaraan}")

File: CPP/Program/ParkingLot.cpp
#pragma once
#include <iostream>
#include <string>
#include <vector>

#include "Motorcycle.cpp"
#include "Car.cpp"

using namespace std;

class ParkingLot
{
private:
    int kapasitas;
    int jml_kendaraan;
    Motorcycle *motorcycle;
    Car *car;

public:
    ParkingLot() {}

    ParkingLot(int kapasitas, int jml_kendaraan)
    {
        this->kapasitas = kapasitas;
        this->jml_kendaraan = jml_kendaraan;
    }

    void setKapasitas(int kapasitas)
    {
        this->kapasitas = kapasitas;
    }

    void setJmlKendaraan(int jml_kendaraan)
    {
        this->jml_kendaraan = jml_kendaraan;
    }

    int getKapasitas()
    {
        return kapasitas;
    }

    int getJmlKendaraan()
    {
        return jml_kendaraan;
    }

    void addVehicle(const vector<Vehicle *> &vehicles)
    {
        int temp = 0;
        for (const auto &vehicle : vehicles)
        {
            temp++;
        }
        if (temp > jml_kendaraan)
        {
            jml_kendaraan = temp;
        }
    }

    void displayInfo()
    {
        cout << "Kapasitas Gedung Parkir: " << kapasitas << endl;
        cout << "Jumlah Kendaraan: " << jml_kendaraan << endl;
    }

    ~ParkingLot() {}
};
